### PR TITLE
🐛 fixes schema-json in settings-CLI

### LINKED ISSUE #2794: settings CLI bugs and improvements
- [x] BUG: schema does not work with new env defaults (
- [x] add --only-set option to display fields that were set. This represents current config (i.e. required+ defaults overriden).
- [x] add --only-required option to display only required fields (minimal config)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

All services apps in the simcore-stack have now a CLI that allows displaying the app settings (associated to its env vars). This can help setup a reliable mechanism to catch up with changes in the configuration between the code and operations (⚠️ devops) 

- 🐛 Fixes problem with app's CLI  command ``settings --as-json-schema``
- ✨Adds flags to get different views (e.g. ``--exclude-unset``)

## Related issue/s

- Closes #2794 

## How to test

- build image, prepare envs and compose specs:
```cmd
make build
make .env
make .stack-simcore-production.yml
```
- get webserver's config values as json
```cmd
docker-compose -f .stack-simcore-production.yml run webserver \
   simcore-service-webserver settings --as-json  > webserver-settings.json
```
- get websercer's config **json-schema**
```cmd
docker-compose -f .stack-simcore-production.yml run webserver \
  simcore-service-webserver settings --as-json-schema  > webserver-settings-schema.json
```

NOTE: this test can also be done deploying the stack and executing ``simcore-service-webserver settings --help`` in the web-server (or any other service) (⚠️ devops).

## Checklist

- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Database migration script? ``cd packages/postgres-database``, ``make setup-commit``, ``sc-pg review -m "my changes"``
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/settings-library/src/settings_library/base.py [hunk 1]
```diff
@@ -106,10 +105,7 @@ def prepare_field(cls, field: ModelField) -> None:
 
     @classmethod
     def create_from_envs(cls, **overrides):
-        # Kept for legacy
+        # Kept for legacy. Identical to the constructor.
         # Optional to use to make the code more readable
         # More explicit and pylance seems to get less confused
-        warnings.warn(
-            "please use constructor instead of `create_from_envs`", DeprecationWarning
-        )
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/base.py:114]: So this is not deprecated?

File: packages/settings-library/src/settings_library/utils_cli.py [hunk 2]
```diff
@@ -11,7 +11,14 @@
 from .base import BaseCustomSettings
 
 
-def print_as_envfile(settings_obj, *, compact: bool, verbose: bool, show_secrets: bool):
+def print_as_envfile(
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/utils_cli.py:14]: missing return type

File: packages/settings-library/src/settings_library/utils_cli.py [hunk 3]
```diff
@@ -37,8 +44,10 @@ def print_as_envfile(settings_obj, *, compact: bool, verbose: bool, show_secrets
         typer.echo(f"{field.name}={value}")
 
 
-def print_as_json(settings_obj, *, compact=False):
-    typer.echo(settings_obj.json(indent=None if compact else 2))
+def print_as_json(settings_obj, *, compact=False, **pydantic_export_options):
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/utils_cli.py:47]: also here

File: packages/settings-library/src/settings_library/base.py [hunk 4]
```diff
@@ -106,10 +105,7 @@ def prepare_field(cls, field: ModelField) -> None:
 
     @classmethod
     def create_from_envs(cls, **overrides):
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/base.py:107]: missing return type

File: packages/settings-library/src/settings_library/utils_cli.py [hunk 5]
```diff
@@ -37,8 +44,10 @@ def print_as_envfile(settings_obj, *, compact: bool, verbose: bool, show_secrets
...
+    typer.echo(
+        settings_obj.json(indent=None if compact else 2, **pydantic_export_options)
+    )
 
 
 def create_settings_command(
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/utils_cli.py:53]: also misses return type

**Review by GitHK — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]
very nice improvement for the settings

File: packages/settings-library/src/settings_library/base.py (hunk 1 above)
Comment (pcrespov) [reply packages/settings-library/src/settings_library/base.py:114]: no

**Pull request merged**